doc/spec.md

```diff
@@ -1498,6 +1498,60 @@ a type reference of the form 'G&lt;A, B>' places no requirements on 'A' but requ
 
 The process of substituting type arguments for type parameters in a generic type or generic signature is known as ***instantiating*** the generic type or signature. Instantiation of a generic type or signature can fail if the supplied type arguments do not satisfy the constraints of their corresponding type parameters.
 
+### Indexed access types { #indexed-access-types }
+
+An indexed access type represents the type of a property, even when the type or the property name is unknown.
+
+&emsp;&emsp;*IndexedAccessType:*  
+&emsp;&emsp;&emsp;*Type*&emsp;`[`&emsp;*Type*&emsp;`]`
+
+
+Indexed access types can be created with the above type syntax `T[K]`, as well as the expression syntax `o[k]`.
+In the preceding example, the indexed type is *T* and the access type is *K*.
+However, neither of these forms is guaranteed to result in an indexed access type; when the indexed type is an object type and the access type is a string type, the type of the property can be determined at the point the indexed access type is requested.
```

Comment:
is a subtype of the string type (have to be careful with `never`)

or we could say that when it's string-like (i.e. string, a string literal, or a union thereof), but we have to be careful about constraints

Reply:
No, the first option is better. :squid::alien::crab: